components/toggle_modal_button_redux/toggle_modal_button_redux.jsx

```diff
@@ -55,12 +58,20 @@ class ModalToggleButtonRedux extends React.PureComponent {
 
         const ariaLabel = formatMessage({id: 'accessibility.button.dialog', defaultMessage: '{dialogName} dialog'}, {dialogName: props.accessibilityLabel});
 
+        let badge = null;
+        if (this.props.modalId === ModalIdentifiers.PLUGIN_MARKETPLACE && !this.props.firstAdminVisitMarketplaceStatus) {
```

Comment:
I still think we should place this logic further up the stack in https://github.com/mattermost/mattermost-webapp/blob/a5b2f799aba060d88c9458c7b03ecbe127898fb5/components/main_menu/main_menu.jsx#L239 and use a `Prop` like `ShowUnread` in this compontent.

Reply:
I'm 1/5 on this. On one side I agree that there is some redundancy/ugliness with the `modalId` check, OTOH, the determination logic is at the same place where it is acted upon so it is more readable/clear imo (don't have to fish up-stack for what conditions determined the badge presence).